src/Compiler/Microsoft.AspNetCore.Razor.Language/test/IntegrationTests/ComponentRenderModeAttributeIntegrationTests.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+
+namespace Microsoft.AspNetCore.Razor.Language.IntegrationTests;
+
+public class ComponentRenderModeAttributeIntegrationTests : RazorIntegrationTestBase
```

Comment:
Why is there a new class just for these three tests?

Reply:
Presumably, we'll want to add more?

Reply:
That's fine. I'm just confused why other tests for rendermode attribute remain in `ComponentCodeGenerationTestBase.cs` and only these three were extracted out of there.

Reply:
It just seemed weird that those tests don't actually assert against baselines, and so don't have matching runtime/design time outputs, so were effectively being run twice for no reason.